llvm/include/llvm/CodeGen/MachineRegisterInfo.h

```diff
@@ -801,6 +801,7 @@ class MachineRegisterInfo {
   /// of an earlier hint it will be overwritten.
   void setRegAllocationHint(Register VReg, unsigned Type, Register PrefReg) {
     assert(VReg.isVirtual());
+    RegAllocHints.grow(Register::index2VirtReg(getNumVirtRegs()));
```

Comment:
Should this be `getNumVirtRegs() - 1`? `grow` expects an index. getNumVRegs() is one past the largest index. Or maybe we could use `resize`?

Reply:
Good catch, opened #102273